### PR TITLE
5.0 target update to from lvalue ternary op

### PR DESCRIPTION
**Pull request by rjenaa**
This test does not pass with LLVM and gcc. 

Clang 18 errors:
:24:41: error: expected addressable lvalue in 'map' clause
:36:31: error: expected addressable lvalue in 'to' clause

gcc 12.2.0 errors:
error: expected ‘)’ before ‘?’ token

### CODE REVIEW

**Comment by fel-cab** [comment]
When ran with GCC 14.1, the error is:
```
test_target_update_to_from_map_lvalue_ternary.cpp:24:47: sorry, unimplemented: unsupported map expression ‘(c ?  a :  b)’
   24 |         #pragma omp target enter data map(to: (c ? a : b))
      |                                               ^
```

File: tests/5.0/target/test_target_update_to_from_map_lvalue_ternary.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	int errors = 0;
+	int before_value;
+	int after_value;
+	int a = -1;
+	int b = 1;
+	bool c = true;
+
+	// Map the variable c ? a : b to the target device
+	// Should evaluate to a
+	#pragma omp target enter data map(to: (c ? a : b))
```
Comment (spophale) [inline tests/5.0/target/test_target_update_to_from_map_lvalue_ternary.cpp:24]:
```suggestion
	#pragma omp target enter data map(to: a,b)
```

File: tests/5.0/target/test_target_update_to_from_map_lvalue_ternary.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			b = 2;
+		}
+	}
+
+	// Copy the updated value from the device to the host variable => b = 2
+	#pragma omp target update from(c ? a : b)
+
+
+	// Exit data region, mapping the variable back from the target device to the host
+	#pragma omp target exit data map(from: c ? a : b)
```
Comment (spophale) [inline tests/5.0/target/test_target_update_to_from_map_lvalue_ternary.cpp:56]:
```suggestion
	#pragma omp target exit data map(delete: a, b)
```

**Comment by spophale** [comment]
@rjenaa will create another test to test ternary operations on target enter/exit data.

**Review by spophale — APPROVED** [review]

**Review by andrewkallai — APPROVED** [review]

**Pull request merged**